### DISCUSSION TITLE
Allow members to delete own member account from front end

### DISCUSSION THREAD

**Discussion by dorothyDorothy** [ExpressionEngine/ExpressionEngine#3007] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
I need to allow members to delete their own accounts (requirement of GDPR) from the front end as I do not want them to have any access to the control panel.

**Describe the solution you'd like**
To be able to use the `public function delete()` in system/ee/ExpressionEngine/Addons/member/mod.member.php

**Teachability, Documentation, Adoption, Migration Strategy**
If you can get this working then I am happy to add it to the docs.

**EE v 6.1.6**
I need to allow members to delete their own accounts (requirement of GDPR) from the front end as I do not want them to have any access to the control panel.
I've found that I can use
`{exp:member:delete}`
but it seems to have lost the location of the language file. Any suggestions of how to fix this?  See screen grab below:

![image](https://user-images.githubusercontent.com/2067584/145098131-3034481c-1942-4dc2-b4d0-7edc793d62a9.png)

The function it is calling is in
system/ee/ExpressionEngine/Addons/member/mod.member.php
`public function delete()`
and I can see the language file with the correct translations here
system/ee/language/english/member_lang.php
but for some reason it is not picking up the language definitions.

I think that the functionality is already in there but just needs some tweaking.

**Comment by eric-lamb62** [comment]
@dorothyDorothy are you saying the only thing wrong with the `delete` tag is the output of the language fields?

**Comment by dorothyDorothy** [comment]
Hi @mithra62 after a discussion on slack with Yuri & Rob, I think that they are right in saying that it is not appropriate to allow members to delete their account from the front end, it is more appropriate that they could move their account to another group for an admin to review and delete and reallocate posts accordingly.

However, there is a problem with the language file not being read properly and some variables being missing.  I think that this should be investigated as it could be an issue with other tags relating to members.
Your assistance is much appreciated.

**Comment by intoeetive** [comment]
For team discussion:
- there is a Role setting that allows member to delete their account; also that functionality is present in legacy templates
- we need to decide what we do with user content; in the CP, there are separate options to anonimize content and delete account

We should also consider "anonimize and delete" option both for CP and frontend

**Comment by eric-lamb62** [comment]
+1 for "anonymize and delete". Content's the tricky part; maybe some logic where if a user has any content they're anonymized and if there's none then they're deleted. 

Also, thinking third party, where content beyond Entries can be a thing, maybe a hook where said Add-ons could "yay or nay" deleted vs anonymized, where if any of them come back "anonymized" removal doesn't happen. Am thinking things like Contact Forms (Forms, Freeform, etc) in this use case and their data integrity.

**Comment by murtaugh** [comment]
+1 for a method that allows front-end users to, at the least, disable their account. 

For my purposes, letting them change their role would be sufficient. I would use a specific role to remove their info from member lists and hide their content from entry loops.
